### DISCUSSION TITLE
Proposal: Int16/Int32/Int64 mark with IInteger interface

### DISCUSSION THREAD

**Discussion by avertiy** [dotnet/runtime#63152] [General]
in addition to .NET 6 – Generic Math i would suggest mark int primitives with a corresponding IInteger interface 
Int16 /  Int32 / Int64 as IInteger interface and 
UInt16 /  UInt32 / UInt64 as IUInteger interface

IInteger interface might combine all interfaces IComparable, IConvertible, IFormattable, IComparable<int>, IEquatable<int>, ISpanFormattable

**Comment by 333fred** [comment]
They [are](https://github.com/dotnet/runtime/blob/84680bf557210114ea5ca823386cd49691c4cac6/src/libraries/System.Private.CoreLib/src/System/Int32.cs#L19).
